modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1307,6 +1307,9 @@ CV__DNN_INLINE_NS_BEGIN
          CV_WRAP void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
                              CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
+     protected:
+         struct DetectionModel_Impl;
+         Ptr<DetectionModel_Impl> impl;
```

Comment:
There is already `Impl` in `Model` class.
No need to keep 2 pointers on "implementations", re-use single one "derived" instance.

Reply:
Replaced to derived Impl usage. Please take a look